ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/extensions/IntSizeExtensions.kt

```diff
@@ -1,6 +1,26 @@
 package com.revenuecat.purchases.ui.revenuecatui.extensions
 
+import android.annotation.SuppressLint
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.onGloballyPositioned
 import androidx.compose.ui.unit.IntSize
+import com.revenuecat.purchases.ui.revenuecatui.data.PaywallState
+import com.revenuecat.purchases.ui.revenuecatui.data.isInFullScreenMode
 
 internal val IntSize.aspectRatio: Float
     get() = width.toFloat() / height
+
+internal val IntSize.isLandscape: Boolean
+    get() = width > height
+
+/**
+ * @return: Modifier with a callback indicating whether landscape layout should be used.
+ */
+@SuppressLint("ModifierFactoryExtensionFunction")
+internal fun PaywallState.Loaded.onLandscapeLayoutChanged(
```

Comment:
Note that changing from/to landscape in a device will trigger a configuration change and recreate the activity, and the whole UI will be recomputed (only the view model will survive). So not sure if this is needed?